forecasttools/idata_w_dates_to_df.py

```diff
@@ -7,80 +7,250 @@
 scoringutils ready dataframes
 """
 
-from datetime import datetime
+from datetime import date, datetime, timedelta
 
 import arviz as az
 import numpy as np
 import polars as pl
+import xarray as xr
 
+import forecasttools
 
-def add_dates_as_coords_to_idata(
-    idata_wo_dates: az.InferenceData,
-    group_dim_dict: dict[str, str],
-    start_date_iso: str,
+
+def is_timedelta_in_days_only(td: timedelta) -> bool:
+    """
+    Checks if a timedelta object can be
+    represented exclusively in days,
+    meaning it has no hours, minutes,
+    seconds, or microseconds.
+    """
+    return td.seconds == 0 and td.microseconds == 0
+
+
+def convert_date_or_datetime_to_np(time_object: any) -> np.datetime64:
+    """
+    Converts a date or datetime object to
+    numpy.datetime64: date -> datetime64[D]
+    (day precision), datetime ->
+    datetime64[ns] (nanosecond precision).
+    """
+    if isinstance(time_object, date) and not isinstance(time_object, datetime):
+        return np.datetime64(time_object, "D")
+    elif isinstance(time_object, datetime):
+        return np.datetime64(time_object, "ns")
+    else:
+        raise TypeError(
+            f"Input must be a date or datetime object; got {type(time_object)}"
+        )
+
+
+def convert_timedelta_to_np(td: timedelta) -> np.timedelta64:
+    """
+    Converts a Python timedelta to:
+    numpy.timedelta64[D] if it is
+    representable in days only. Otherwise,
+    convert to numpy.timedelta64[ns].
+    """
+    if is_timedelta_in_days_only(td):
+        return np.timedelta64(td.days, "D")
+    else:
+        return np.timedelta64(int(td.total_seconds() * 1e9), "ns")
+
+
+def generate_time_range_for_dim(
+    start_time_as_dt: datetime | date,
+    variable_data: xr.DataArray,
+    dimension: str,
+    time_step: timedelta,
+) -> list[date] | list[datetime]:
+    """
+    Generates a range of times based on the
+    start date, time step, and variable's
+    dimension size. A range of dates is
+    generated if the start date is a date and
+    the time step is in days. A range of times
+    is generated if the start date is a time
+    and/or the time step is a time.
```

Comment:
Remove docstring text that is no longer accurate
```suggestion
    dimension size. 
```